### PR TITLE
works on higher function and jasmine

### PR DESCRIPTION
**Pull request by tfb414**
@radishmouse please review

### CODE REVIEW

File: DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+function isPositive(n){
+    return n > 0;
+}
+function isEven(n){
+    return n % 2 === 0;
+}
+function squareIt(n){
+    return n * n;
+}
+function coolCity(obj){
```
Comment (radishmouse) [inline DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js:11]: Consider accepting threshold temp as argument.

Comment (radishmouse) [inline DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js:11]: Maybe rename function to something like "isTempLowerThan"

File: DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    return obj.temperature < 70;
+}
+function returnCityName(obj){
```
Comment (radishmouse) [inline DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js:14]: Maybe rename to "getName" (not a dealbreaker by any means)

File: DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+function returnTotal(array){
+    return array.map(returnPrice).reduce((a,b)=> a + b);
+
+}
+
+function returnPrice(obj){
+    return obj.price
+}
+function addSep(array, sep){
+    function derp(currentValue, index, array){
```
Comment (radishmouse) [inline DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js:32]: This could use a refactor. Compare to V2 below.

Comment (tfb414) [reply DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js:32]: I did an additional exercise to refactor this and left the original as an example of how to make things overly complicated

**Comment by tfb414** [comment]
@radishmouse I made the changes that you suggested....I think!

**Comment by radishmouse** [comment]
Hey there! Did you commit & push those changes to the same branch? (`tfb414/code-practice`)

**Comment by radishmouse** [comment]
(I ask because github is only showing 1 commit for this PR)

**Comment by tfb414** [comment]
It's possible that I got my branches mixed up. Let me look at it and get
back to you.

On Jul 18, 2017 6:15 AM, "Chris Aquino" <notifications@github.com> wrote:

(I ask because github is only showing 1 commit for this PR)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/tfb414/code-practice/pull/1#issuecomment-316020435>,
or mute
the thread

**Pull request merged**